examples/Spirit/spirit_utils.cc

```diff
@@ -164,28 +166,19 @@ std::tuple<
   return {std::move(modeVector), std::move(toeEvals), std::move(toeEvalSets)};
   // return {std::move(toeEvals), std::move(toeEvalSets)};
 }
-// //Overload function to allow the use of a equal number of knotpoints for every mode.
-// template <typename T>
-// std::tuple<
-//                 DirconModeSequence<T>,
-//                 std::vector<std::unique_ptr<multibody::WorldPointEvaluator<T>>> ,
-//                 std::vector<std::unique_ptr<multibody::KinematicEvaluatorSet<T>>>
-//           > createSpiritModeSequence( 
-//   MultibodyPlant<T>& plant, // multibodyPlant
-//   Eigen::Matrix<bool,-1,4> modeSeqMat, // bool matrix describing toe contacts as true or false e.g. {{1,1,1,1},{0,0,0,0}} would be a full support mode and flight mode
-//   uint16_t knotpoints, // Number of knot points per mode
-//   double mu = 1){
-//   int numModes = modeSeqMat.rows(); 
-//   std::vector<int> knotpointVect = Eigen::MatrixXi::Constant(numModes,1,knotpoints);
-//   return createSpiritModeSequence(plant, modeSeqMat,knotpointVect, mu);
-// }
-
-
-
-
 
 
+template <typename T>
+std::tuple<
+                std::vector<std::unique_ptr<DirconMode<T>>>,
+                std::vector<std::unique_ptr<dairlib::multibody::WorldPointEvaluator<T>>> ,
+                std::vector<std::unique_ptr<dairlib::multibody::KinematicEvaluatorSet<T>>>
+          > createSpiritModeSequence( 
+          MultibodyPlant<T>& plant, // multibodyPlant
+          dairlib::ModeSequenceHelper* msh ){
```

Comment:
Consider changing the ptr to a const reference since the msh won't be changed and we don't need it to be null.

Reply:
good idea